lib/transactions-common.js

```diff
@@ -1426,8 +1426,10 @@ Transact.prototype._processTransaction = function (txid, description, items, con
   
   // First, need to iterate over the changes that are going to be made and make sure that,
   // if there are hard removes, the db version of the doc gets stored on the transaction
+  // We only do this step if the remove is not instant - otherwise the doc would already have
+  // been removed from the collection by this point.   
   _.each(items, function (item, index) {
-    if (item.action === "remove" && item.hardDelete) {
+    if (item.action === "remove" && item.hardDelete && !item.instant) {
       // Get the existing doc and store it in the transaction record
       // We overwrite the temporary version of the doc from an instant remove on the client
       // Because chances are that the whole document was not available on the client
```

Comment:
> I think it's acknowledged in the docs that instant client side removes could result in data corruption/loss due to unpublished fields.

[Confirmed](https://github.com/JackAdams/meteor-transactions/blame/f15047113a60a796aec5679db764d92c5f130610/README.md#L180), so the comment about instant removes from the client can just be removed.